src/palladio/SOPAssign.cpp

```diff
@@ -1005,7 +1006,8 @@ void SOPAssign::updatePrimitiveAttributes(GU_Detail* detail) {
 								UT_String result;
 								if (choices->tokenFromIndex(result, intValue)) {
 									GA_RWHandleS stringHandle(detail->addStringTuple(attrOwner, attributeName, 1));
-									stringHandle.set(0, result);
+									for (GA_Iterator it(primitiveRange); !it.atEnd(); ++it)
+										stringHandle.set(it.getOffset(), result);
```

Comment:
Consider deduplicating all these loops with something like this:
```
template<typename H, typename V>
void setAttributeRange(GA_Range range, H handle, V value) {
	for (GA_Iterator it(range); !it.atEnd(); ++it)
		handle.set(it.getOffset(), value);
}

```

Reply:
fixed